### PR TITLE
mps3-an547:support nuttx run qemu(cortex-m55)

### PR DESCRIPTION
**Pull request by anjiahao1**
## Summary
support nuttx run qmeu
## Impact
mps3-an547
## Testing

```
   Configuring NuttX and compile:
   $ ./tools/configure.sh -l mps3-an547:nsh
   $ make
   Running with qemu
   $ qemu-system-arm -M mps3-an547 -nographic -kernel nuttx.bin

```

run `ostest` pass

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/include/mps/irq.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+/****************************************************************************
+ * Pre-processor Prototypes
+ ****************************************************************************/
+
+#define NR_IRQS                 130  /* Total number of interrupts */
```
Comment (anchao) [inline arch/arm/include/mps/irq.h:36]: why 130

File: drivers/serial/serial_cmsdk.c [hunk 2]
```diff
@@ -0,0 +1,796 @@
...
+ *
+ * Description:
+ *   All ioctl calls will be routed through this method
+ *
+ ****************************************************************************/
+
+static int uart_cmsdk_ioctl(struct file *filep, int cmd, unsigned long arg)
+{
+  FAR struct uart_dev_s *dev = filep->f_inode->i_private;
+  FAR struct uart_cmsdk_s *priv = (FAR struct uart_cmsdk_s *)dev->priv;
```
Comment (anchao) [inline drivers/serial/serial_cmsdk.c:500]:
```suggestion
  FAR struct uart_cmsdk_s *priv = dev->priv;
```
remove all `void *` cast

File: boards/arm/mps/mps3-an547/include/board.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __BOARDS_ARM_MSP_MSP3_AN547_INCLUDE_BOARD_H
```
Comment (anchao) [inline boards/arm/mps/mps3-an547/include/board.h:21]:
```suggestion
#ifndef __BOARDS_ARM_MPS_MPS3_AN547_INCLUDE_BOARD_H
```

**Review by acassis — COMMENTED** [review]
Please fix commit title: qmeu -> qemu

File: arch/arm/include/mps/irq.h (hunk 1 above)
Comment (anjiahao1) [reply arch/arm/include/mps/irq.h:36]: image
[mps3-an547.pdf](https://developer.arm.com/-/media/Arm%20Developer%20Community/PDF/DAI0547B_SSE300_PLUS_U55_FPGA_for_mps3.pdf)

page 38 Table 7-1

**Pull request merged**